source/extensions/filters/http/jwt_authn/extractor.h

```diff
@@ -0,0 +1,118 @@
+#pragma once
+
+#include <map>
+#include <string>
+#include <unordered_set>
+
+#include "envoy/config/filter/http/jwt_authn/v2alpha/config.pb.h"
+
+#include "common/common/logger.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace JwtAuthn {
+
+/**
+ * JwtLocation stores following info
+ * *  extracted JWT string,
+ * *  the location where the JWT is extracted from,
+ * *  list of issuers specified the location. The issuer of extracted JWT must
+ * match one of these
+ * issuers.
+ */
+class JwtLocation {
+public:
+  virtual ~JwtLocation() {}
+
+  // Get the token string
+  virtual const std::string& token() const PURE;
+
+  // Check if an issuer has specified the location.
+  virtual bool isIssuerSpecified(const std::string& issuer) const PURE;
+
+  // Remove the token from the headers
+  virtual void removeJwt(Http::HeaderMap& headers) const PURE;
+};
+
+typedef std::unique_ptr<JwtLocation> JwtLocationPtr;
+
+/**
+ * Extracts JWT from locations specified in the config.
+ *
+ * The rules of JWT extraction:
+ * * Each issuer can specify its locations either at headers or query
```

Comment:
either or either/or? I.e. it looks like you can specify both if you want..

Reply:
Thanks, updated the comment to support both